api/matrix_lib.cpp

```diff
@@ -0,0 +1,185 @@
+#include "matrix_lib.h"
+
+#include <algorithm>
+#include <iostream>
+
+// Include C header file
+extern "C" {
+#include "matrix_operation.h"
+}
+
+namespace MatrixLib {
+
+Matrix::Matrix(const size_t rows, const size_t cols) {
+    if (rows == 0 || cols == 0) {
+        throw std::invalid_argument("Size must be positive");
```

Comment:
In future we need get rid of std::<something> as exception and use fully custom ones
it is needed for user to understand that exactly this lib throw an exception
see for example: https://www.geeksforgeeks.org/how-to-throw-custom-exception-in-cpp/